.github/workflows/tests.yml

```diff
@@ -27,13 +27,74 @@ jobs:
       - name: Rebase to current master
         run: git rebase origin/master
 
+      - name: Prepare
+        run: |
+          # # override GitHub's bind mount from host, we don't want anything from there and it interferes with ssh
+          # export HOME=$(getent passwd $(id -u) | cut -f6 -d:)
+
+          # Put the /secrets in place which we'll mount to the tasks-container
+          mkdir -p ~/secrets
+          echo '${{ secrets.COCKPITUOUS_TOKEN }}' > ~/.config/github-token
+          echo '${{ secrets.SSH_KNOWN_HOSTS }}' > ~/secrets/known_hosts
+          echo '${{ secrets.SINK_SSH_CONFIG }}' > ~/secrets/ssh-config
+          echo '${{ secrets.SINK_ID_RSA }}' > ~/secrets/id_rsa
+          echo '${{ secrets.SINK_ID_RSA_PUB }}' > ~/secrets/id_rsa.pub
+          sudo chmod 400 ~/secrets/*
+          # change permissions to user in cockpit-tasks
+          # TODO sudo needed?
+          sudo chown 1111:1111 ~/secrets/*
+
+          # install dependencies needed to use bots stuff in the host container
+          sudo pip3 install pika
+
+      # - name: Setup tmate session
+      #   uses: mxschmitt/action-tmate@v3
+      #   timeout-minutes: 45
+
       - name: Build and test tasks container if it changed
         run: |
           changes=$(git diff --name-only origin/master..HEAD -- tasks/)
           if [ -n "${changes}" ]; then
               # Run podman as root, as podman is missing slirp4netns by default, and does not have overlayfs by default
-              sudo make tasks-container
-              # This does not do much yet, just iterate to the AMQP queue 30 times and exit
-              # TODO: Inject a workload that does not need /dev/kvm
-              sudo tasks/run-local.sh
+              # TODO uncomment, make it faster now
+              # sudo make tasks-container
+
+              # Generate the secrets in tree and copy the working github token
+              (cd tasks/credentials; ./generate-ca.sh)
+              (cd tasks/credentials/webhook; ./generate.sh)
+              chmod -R go+rX tasks/credentials/webhook
```

Comment:
Wait, what does that do? run-local.sh already generates secrets, why does it need to happen again here? This feels too complex for the workflow file.

Reply:
Yea I wasn't sure how to solve this. `run-local` won't generate them if it thinks they're already in the tree, and I need them in a predictable location, so I can copy in the webhook token before running the tasks container.

Reply:
Ah! In fact I [have a commit](https://github.com/martinpitt/cockpituous/commit/ccd5639aa7a40bc12df85f6b44aa5bf36541e7e3) to put them into a static path. I just didn't propose it yet as I didn't want to interfere with your work too much. But maybe that one should go in then? (Feel free to just cherry-pick it)